### ISSUE TITLE
Crash onBackPressed if no image selected

### ISSUE THREAD

**Issue by andreealaura16** [esafirm/android-image-picker#186]
If I want to go back to my view, without selecting any image, the app just terminates. Same problem for single or multi selection. 

Any idea why this is and how to solve it?

**Comment by aocet** [comment]
@Override
    public void onActivityResult(int requestCode, final int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (ImagePicker.shouldHandle(requestCode, resultCode, data)) {
            images = (ArrayList<Image>) ImagePicker.getImages(data);
            File imgFile = new File(images.get(0).getPath());
            if(imgFile.exists()){
                compress(imgFile);
                uploadImage(imgFile);
                refresh();
            }
            return;
        }
        super.onActivityResult(requestCode, resultCode, data);
    }

Dont forget line 4

**Comment by CasperVerswijvelt** [comment]
Having the same issue here. When debugging I see that the 'onActivityResult' method's body never runs, since the 'data' parameter is null, when the method declaration says it can't be null. I am using Kotlin for this project, as opposed to java being used in the examples tho. Maybe this has something to do with Kotlin's nullable types and null safety that isn't really present in Java?

<details><summary>Show error</summary>
<p>

```kotlin
`2018-12-29 14:01:31.925 11356-11356/be.verswijvelt.casper.beerio E/AndroidRuntime: FATAL EXCEPTION: main
    Process: be.verswijvelt.casper.beerio, PID: 11356
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=66089, result=0, data=null} to activity {be.verswijvelt.casper.beerio/be.verswijvelt.casper.beerio.MainActivity}: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter data
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4496)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4538)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:49)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1906)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6863)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter data
        at be.verswijvelt.casper.beerio.fragments.AddBeerFragment.onActivityResult(Unknown Source:37)
        at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:160)
        at android.app.Activity.dispatchActivityResult(Activity.java:7476)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4489)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4538) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:49) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1906) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6863) `
```

</p>
</details>

**Comment by CasperVerswijvelt** [comment]
An updatehere, I tried to use other libraries for image picker to try and fix this, to come upon the same bug in them aswell. The problem was that I was doing all of this trough a fragment, and only overrid the 
'onActivityResult(requestCode: Int, resultCode: Int, data: Intent)' method. If data was null, it would crash. 

I was able to fix this by overriding the 'onActivityResult(requestCode: Int, resultCode: Int, data: Intent?)' method in the **parent activity** (note that here data IS a nullable parameter) with this implementation:

```Kotlin
override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
    if(data == null) return
    super.onActivityResult(requestCode, resultCode, data)
}
```

Note that I did this fix to fix it with another image picker library, but i suspect the fix will be the same for this one. I don't know if this is the right method, but it works for me for now.
